python-api/livy/client.py

```diff
@@ -0,0 +1,444 @@
+#
+# Licensed to Cloudera, Inc. under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  Cloudera, Inc. licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from ConfigParser import SafeConfigParser
+from StringIO import StringIO
+import os
+from concurrent.futures import ThreadPoolExecutor
+import json
+import cloudpickle
+import base64
+from job_handle import JobHandle
+import re
+from urlparse import ParseResult
+import traceback
+import requests
+import threading
+
+
+class HttpClient(object):
+    """A http based client for submitting Spark-based jobs to a Livy backend.
+
+    Parameters
+    ----------
+    uri : urlparse.ParseResult instance
+        Livy server uri generated from urlparse lib
+    load_defaults : boolean, optional
+        This parameter decides if the default config needs to be loaded
+        Default is True
+    conf_dict : dict, optional
+        The key-value pairs in the conf_dict will be loaded to the config
+
+    Examples
+    --------
+    Imports needed to create an instance of HttpClient
+    >>> from livy.client import HttpClient
+    >>> from urlparse import urlparse
+
+    1) Creates a client that is loaded with default config
+       as 'load_defaults' is True by default
+    >>> client = HttpClient(urlparse("http://example:8998/"))
```

Comment:
This seems a little odd, since the only way to create a `ParseResult` is to call `urlparse`; so it might be better to just take a string as input.